packages/modules/inventory-next/src/services/inventory.ts

```diff
@@ -799,33 +800,68 @@ export default class InventoryModuleService<
    * @return The updated inventory level
    * @throws when the inventory level is not found
    */
-  @InjectManager("baseRepository_")
-  @EmitEvents()
-  async adjustInventory(
+  adjustInventory(
     inventoryItemId: string,
     locationId: string,
     adjustment: number,
+    context: Context
+  ): Promise<InventoryNext.InventoryLevelDTO>
+
+  adjustInventory(
+    data: {
+      inventoryItemId: string
+      locationId: string
+      adjustment: number
+    }[],
+    context: Context
+  ): Promise<InventoryNext.InventoryLevelDTO[]>
+
+  @InjectManager("baseRepository_")
+  @EmitEvents()
+  async adjustInventory(
+    inventoryItemIdOrData: string | any,
+    locationId?: string | Context,
+    adjustment?: number,
     @MedusaContext() context: Context = {}
-  ): Promise<InventoryNext.InventoryLevelDTO> {
-    const result = await this.adjustInventory_(
-      inventoryItemId,
-      locationId,
-      adjustment,
-      context
-    )
+  ): Promise<
+    InventoryNext.InventoryLevelDTO | InventoryNext.InventoryLevelDTO[]
+  > {
+    let all: any = inventoryItemIdOrData
+
+    if (isString(inventoryItemIdOrData)) {
+      all = [
+        {
+          inventoryItemId: inventoryItemIdOrData,
+          locationId,
+          adjustment,
+        },
+      ]
+    }
 
-    context.messageAggregator?.saveRawMessageData({
-      eventName: InventoryEvents.inventory_level_updated,
-      metadata: {
-        service: this.constructor.name,
-        action: CommonEvents.UPDATED,
-        object: "inventory-level",
-      },
-      data: { id: result.id },
-    })
+    const results: TInventoryLevel[] = []
+
+    for (const data of all) {
+      const result = await this.adjustInventory_(
+        data.inventoryItemId,
+        data.locationId,
+        data.adjustment,
+        context
+      )
+      results.push(result)
+
+      context.messageAggregator?.saveRawMessageData({
+        eventName: InventoryEvents.inventory_level_updated,
+        metadata: {
+          service: this.constructor.name,
```

Comment:
discussion: I ve seen that in many places, the event service is the constructor name of the class, but on my side I use the Modules.MY_MODULE instead. I think it should be the module identifier rather than the class name cause in a distributed system I believe the module (e.g microservice) name is more relevant than the underlying service class name which can have a random name. wdyt? dont change it I ll do it as part of the event task

Reply:
Yes, it has to be the Module definition key.
I didn't even pay attention to these events in this PR :D

Reply:
ahah no worries, I started to change them in another pr

Reply:
cc @olivermrbl just for info